### PR TITLE
AK+Kernel: More OOM hardening,  and removal of allocations with new StringView::for_each_split_view API

### PR DESCRIPTION
**Pull request by bgianfo**
More work to chip away at #6369

### CODE REVIEW

File: Kernel/Arch/x86/ProcessorInfo.h [hunk 1]
```diff
@@ -32,7 +33,7 @@ class ProcessorInfo {
 
     const String& cpuid() const { return m_cpuid; }
     const String& brandstr() const { return m_brandstr; }
-    const String& features() const { return m_features; }
+    StringView features() const { return m_features->view(); }
```
Comment (creator1creeper1) [inline Kernel/Arch/x86/ProcessorInfo.h:36]: Would it also make sense to convert the functions above this to `StringView`? `String const&` with the double-indirection seems generally undesirable to me.

File: Kernel/Arch/x86/ProcessorInfo.h [hunk 2]
```diff
@@ -20,7 +21,7 @@ class ProcessorInfo {
     Processor& m_processor;
     String m_cpuid;
     String m_brandstr;
```
Comment (creator1creeper1) [inline Kernel/Arch/x86/ProcessorInfo.h:23]: Can we make these `KString` as well?

File: AK/StringView.cpp [hunk 3]
```diff
@@ -36,24 +36,8 @@ StringView::StringView(const ByteBuffer& buffer)
...
-            if (sublen != 0 || keep_empty)
-                v.append(substring_view(substart, sublen));
-            substart = i + 1;
-        }
-    }
-    size_t taillen = length() - substart;
-    if (taillen != 0 || keep_empty)
-        v.append(substring_view(substart, taillen));
-    return v;
+    StringView seperator_view { &separator, 1 };
```
Comment (creator1creeper1) [inline AK/StringView.cpp:39]: Isn't this pretty inefficient? Splitting by a `StringView` seems a lot more computationally complex than splitting by a `char`.

File: AK/StringView.h [hunk 4]
```diff
@@ -120,6 +120,36 @@ class StringView {
 
     [[nodiscard]] Vector<StringView> split_view_if(Function<bool(char)> const& predicate, bool keep_empty = false) const;
 
+    template<VoidFunction<StringView> Callback>
+    void for_each_split_view(char separator, bool keep_empty, Callback callback) const
+    {
+        StringView seperator_view { &separator, 1 };
```
Comment (creator1creeper1) [inline AK/StringView.h:126]: See above.

File: AK/StringView.cpp [hunk 5]
```diff
@@ -42,27 +42,10 @@ Vector<StringView> StringView::split_view(const char separator, bool keep_empty)
...
-        if (keep_empty || separator_index > 0)
-            parts.append(part_with_separator.substring_view(0, separator_index));
-        view = view.substring_view_starting_after_substring(part_with_separator);
-        maybe_separator_index = view.find(separator);
-    }
-    if (keep_empty || !view.is_empty())
+    for_each_split_view(separator, keep_empty, [&](StringView view) {
         parts.append(view);
-
+    });
```
Comment (creator1creeper1) [inline AK/StringView.cpp:48]: Neat implementation.

File: Kernel/Arch/x86/ProcessorInfo.h (hunk 1 above)
Comment (bgianfo) [reply Kernel/Arch/x86/ProcessorInfo.h:36]: Yes, but that's for another day.

File: Kernel/Arch/x86/ProcessorInfo.h (hunk 2 above)
Comment (bgianfo) [reply Kernel/Arch/x86/ProcessorInfo.h:23]: Yes, bit I chose to carve off this piece for now. There are tons of things to fix, and not so much time. 😄

File: AK/StringView.cpp (hunk 3 above)
Comment (bgianfo) [reply AK/StringView.cpp:39]: I guess I can write a benchmark to compare before and after.

Comment (awesomekling) [reply AK/StringView.cpp:39]: Either way, seperator => separator :^)

**Pull request merged**